crates/astria-core/src/slinky/types.rs

```diff
@@ -0,0 +1,94 @@
+pub mod v1 {
+    use crate::generated::astria_vendored::slinky::types::v1 as raw;
+
+    #[cfg_attr(
+        feature = "serde",
+        derive(serde::Deserialize, serde::Serialize),
+        serde(from = "raw::CurrencyPair", into = "raw::CurrencyPair")
+    )]
+    #[derive(Debug, Clone, PartialEq, Eq, Hash)]
+    pub struct CurrencyPair {
+        base: String,
+        quote: String,
+    }
+
+    impl CurrencyPair {
+        #[must_use]
+        pub fn base(&self) -> &str {
+            &self.base
+        }
+
+        #[must_use]
+        pub fn quote(&self) -> &str {
+            &self.quote
+        }
+
+        #[must_use]
+        pub fn from_raw(raw: raw::CurrencyPair) -> Self {
+            Self {
+                base: raw.base,
+                quote: raw.quote,
+            }
+        }
+
+        #[must_use]
+        pub fn into_raw(self) -> raw::CurrencyPair {
+            raw::CurrencyPair {
+                base: self.base,
+                quote: self.quote,
+            }
+        }
+    }
+
+    impl From<raw::CurrencyPair> for CurrencyPair {
+        fn from(raw: raw::CurrencyPair) -> Self {
+            Self::from_raw(raw)
+        }
+    }
+
+    impl From<CurrencyPair> for raw::CurrencyPair {
+        fn from(currency_pair: CurrencyPair) -> Self {
+            currency_pair.into_raw()
+        }
+    }
+
+    impl std::fmt::Display for CurrencyPair {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+            write!(f, "{}/{}", self.base, self.quote)
+        }
+    }
+
+    impl std::str::FromStr for CurrencyPair {
+        type Err = CurrencyPairParseError;
+
+        fn from_str(s: &str) -> Result<Self, Self::Err> {
+            s.split_once('/')
```

Comment:
Can you please use a regex for this instead? Something like `([a-zA-Z]+)/([a-zA-Z])` two define two groups - or just `alnum` if it can contain digits too.

Does slinky define what can be included in these pairs anywhere?

Reply:
the pairs themselves aren't defined anywhere afaik, but they should be two alphabetical strings with a `/` between.